be2-scala/src/main/scala/ch/epfl/pop/model/objects/ElectionChannel.scala

```diff
@@ -20,7 +20,7 @@ object ElectionChannel {
      * @tparam T type of data to extract from the election channel
      * @return Future of a list of tuple containing the message and the data extracted
      */
-    def extractMessages[T: Manifest](dbActor: AskableActorRef = DbActor.getInstance): Future[List[(Message, T)]] = {
+    def extractMessages[T: reflect.ClassTag](dbActor: AskableActorRef = DbActor.getInstance): Future[List[(Message, T)]] = {
```

Comment:
This generic type T was extending Manifest to accept all case class. We could now make it accept all case class extendding MessageData 
def extractMessages[T: MessageData]
for readability

Reply:
T does not extend `Manifest` either way. This is the syntax for passing an implicit `Manifest[T]` to allow pattern matching to check for instances of `T`. Here the only difference it that `Manifest` is gone in Scala 3. Orthogonally to that, I agree that we could specify `T <: MessageData`, but that won't replace `Manifest`.

Reply:
Can you review #1113 instead? That would be more practical